### PR TITLE
Add CLA signing and verification

### PR DESCRIPTION
**Pull request by cb1kenobi**
This PR adds the CLA form (or link to download signed CLA) to the Contributors page. In addition, it also provides a REST API endpoint for the CLA status check to verify that the user has signed the CLA. The signed PDF and metadata files are stored on Google Cloud Storage.

image

image

This PR also updates dependencies and cleans up global styles.

### CODE REVIEW

File: pages/index.tsx [hunk 1]
```diff
@@ -1,9 +1,9 @@
-import Head from 'next/head';
-import Image from 'next/image';
+// import Head from 'next/head';
+// import Image from 'next/image';
```
Comment (m1ga) [inline pages/index.tsx:2]:
```suggestion
```

remove if not needed

File: pages/index.tsx [hunk 2]
```diff
@@ -1,9 +1,9 @@
...
 import Hero from '../components/home/hero';
 import About from '../components/home/about';
-import Team from '../components/home/team';
-import Companies from '../components/home/companies';
+// import Team from '../components/home/team';
+// import Companies from '../components/home/companies';
```
Comment (m1ga) [inline pages/index.tsx:6]:
```suggestion
```

remove if not needed

File: pages/blog/index.tsx [hunk 3]
```diff
@@ -23,7 +23,7 @@ export default function Blog(props: InferGetStaticPropsType<typeof getStaticProp
                             </div>
                             <h2 className='text-lg font-bold sm:text-xl md:text-2xl'>
                                 <Link href={`/blog/${page.slug}`}>
-                                    <a>{page.title}</a>
+                                    {page.title}
                                 </Link>
                             </h2>
                             <p className='text-sm text-gray-500'>{page.teaser}…</p>
```
Comment (m1ga) [inline pages/blog/index.tsx:29]:
```suggestion
                            <p className='text-sm'>{page.teaser}…</p>
```
make blog teaser text white again

Comment (m1ga) [reply pages/blog/index.tsx:29]: and please make the next line
```
<p className='pt-2 text-xs font-medium text-gray-500'>
```
can't add a suggestion there.
Screenshot_20230916_152919

File: utils/cla.ts [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+		});
+	};
+
+	// page.drawRectangle({
+	// 	x: 80,
+	// 	y: pageHeight - 188,
+	// 	width: 210,
+	// 	height: 24,
+	// 	color: rgb(1, 0, 0)
+	// });
```
Comment (m1ga) [inline utils/cla.ts:150]:
```suggestion
```

remove

File: utils/cla.ts [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+	drawText(80, 422, 'tidev@smalltownhosting.com');
+	drawText(80, 480, today);
+
+	// page.drawRectangle({
+	// 	x: 316,
+	// 	y: pageHeight - 188,
+	// 	width: 210,
+	// 	height: 34,
+	// 	color: rgb(1, 0, 0)
+	// });
```
Comment (m1ga) [inline utils/cla.ts:164]:
```suggestion
```

remove

File: utils/cla.ts [hunk 6]
```diff
@@ -0,0 +1,208 @@
...
+
+	pdfDoc.setTitle('TiDev Contributor License Agreement');
+	pdfDoc.setAuthor('TiDev, Inc.');
+	pdfDoc.setCreator('https://tidev.io');
+	pdfDoc.setModificationDate(new Date());
+
+	const unsignedBytes = await pdfDoc.save();
+	const unsignedFile = join(tmpdir(), `${githubUsername}.unsigned.pdf`);
+	const signedFile = join(tmpdir(), `${githubUsername}.pdf`);
+	// const signedFile = `/home/chris/projects/titanium/tidev/www/${githubUsername}.pdf`;
```
Comment (m1ga) [inline utils/cla.ts:181]:
```suggestion
```
remove

File: utils/cla.ts [hunk 7]
```diff
@@ -0,0 +1,208 @@
...
+			`--image ${signatureFile}`,
+			'--top 22.5',
+			'--left 3.5',
+			'--width 14'
+		);
+	} finally {
+		await unlink(unsignedFile);
+	}
+
+	// throw new Error('done!');
```
Comment (m1ga) [inline utils/cla.ts:205]:
```suggestion
```

remove

**Review by m1ga — COMMENTED** [review]
left some comments

File: utils/cla.ts (hunk 7 above)
Comment (cb1kenobi) [reply utils/cla.ts]: Done.

File: utils/cla.ts (hunk 6 above)
Comment (cb1kenobi) [reply utils/cla.ts]: Done.

File: utils/cla.ts (hunk 4 above)
Comment (cb1kenobi) [reply utils/cla.ts:159]: These serve a purpose. I added a comment to explain their use.

File: pages/index.tsx (hunk 2 above)
Comment (cb1kenobi) [reply pages/index.tsx:4]: Eh, they aren't harming anything and they link to components that do exist, but were never finished. I'd rather leave them alone as a reminder of where to add them back someday.

Comment (m1ga) [reply pages/index.tsx:4]: sounds good

**Comment by m1ga** [comment]
Just a quick though:

Maybe add another div around the pages in contribute.tsx line195 and make it `height:700px, overflow:auto;border-radius:5px` ? 

Screenshot_20230917_114741

**Comment by cb1kenobi** [comment]
I tried setting a height and it was horrible on mobile.

**Comment by m1ga** [comment]
you can use something like `80vh` so it will be 80% of the viewport height in combination with `min-height: 500px` to give it a minimum in case you use your phone horizontal.

**Comment by cb1kenobi** [comment]
I don't agree with the border radius. It's unnecessary.

I don't agree with the capping the height. Like I implied, I already went down this road. Scrolling complicates UX. it complicates printing. It complicates the mobile experience. I wouldn't be surprised if if the scrolling complicated the overlayed field scaling. Remember, this is an agreement page. It's not a PDF viewer.

**Comment by m1ga** [comment]
you are right, was overthinking it :smile: Should be quickly for the user and not make more steps then needed :+1:

**Comment by cb1kenobi** [comment]
Here's the PR for the CLA action that uses the new API endpoint: https://github.com/tidev/tidev-cla-action/pull/3.

**Comment by cb1kenobi** [comment]
Let's go!

**Pull request merged**